includes/discovery/vlans/q-bridge-mib.inc.php

```diff
@@ -37,6 +37,20 @@
         }, []);
     }
 
+    //if 'VlanStaticName' is not set, LNMS won't discover Vlan although Vlan is functional on device
+    foreach ($untag as $vId => $tmp) {
+        if (! $vlans[$vtpdomain_id]['dot1qVlanStaticName']) {
+            $vlans[$vtpdomain_id] = ['dot1qVlanStaticName' => 'Vlan_' . $vId];
+            d_echo("Vlans: assigned generic Vlan_$vId name");
+        }
+    }
+    foreach ($tagoruntag as $vId => $tmp) {
+        if (! $vlans[$vtpdomain_id]['dot1qVlanStaticName']) {
+            $vlans[$vtpdomain_id] = ['dot1qVlanStaticName' => 'Vlan_' . $vId];
+            d_echo("Vlans: assigned generic Vlan_$vId name");
+        }
+    }
+
     foreach ($vlans as $vlan_id => $vlan) {
```

Comment:
Need to change this to loop over the other arrays instead of $vlans.

Reply:
i don't understand

    foreach ($vlans as $vlan_id => $vlan) {
will not reach missing ID because it not exist in $vlans array
... and later
    $vlan_upd['vlan_name'] = $vlan['dot1qVlanStaticName'];
again, $vlans array dot1qVlanStaticName is used

so i could not imagine how to solve this in other way
only to fill $vlans array with missing data before reach 
    foreach ($vlans as $vlan_id => $vlan) {

Reply:
DBG before: 
Array
(
    [2] => Array
        (
            [dot1qVlanStaticName] => guest
        )

    [254] => Array
        (
            [dot1qVlanStaticName] => mgmn
        )

    [518] => Array
        (
            [dot1qVlanStaticName] => mcast
        )

)

Vlans: assigned generic Vlan_1 name  

DBG after: 
Array
(
    [2] => Array
        (
            [dot1qVlanStaticName] => guest
        )

    [254] => Array
        (
            [dot1qVlanStaticName] => mgmn
        )

    [518] => Array
        (
            [dot1qVlanStaticName] => mcast
        )

    [1] => Array
        (
            [dot1qVlanStaticName] => Vlan_1
        )

)

Reply:
Yeah, the VLAN code is very hard to follow.  Unfortunately, your fix makes it harder to follow.  

The easiest thing would be to combine all the data into one array during the snmp calls then we only need one foreach loop.


```php
[
  1 => array:1 [
    "dot1qVlanStaticName" => "default",
    "dot1qVlanCurrentUntaggedPorts" => "FF C0 00 00 00 00 00 00 03 FF 80 00 00"
    "dot1qVlanCurrentEgressPorts" => "FF C0 00 00 00 00 00 00 03 FF 80 00 00"
  ]
]
```

Reply:
Then if dot1qVlanStaticName is not defined for that vlan id, we can then generate a name.

Reply:
ok @murrant 
i made a new q-bridge-mib as you wish, completly rewritten, using new Laravel/DB etc
but i will wait what will happen with Eltex mes21xx/23xx PR to avoid test/data/json problems
after that i will send PR